lib/soft_deletion/core.rb

```diff
@@ -86,11 +86,25 @@ def soft_delete_dependencies
 
     protected
 
-    def update_soft_delete_counter_caches(value)
-      each_counter_cached_associations do |association|
-        association.load_target unless association.loaded?
-        if (target = association.target) # uncovered TODO: add test coverage for target not existing
-          target.class.update_counters(target.id, association.reflection.counter_cache_column => value)
+    if ActiveRecord.version >= Gem::Version.new("7.2.0")
+      def update_soft_delete_counter_caches(value)
+        counter_cached_association_names.each do |association_name|
+          association = association(association_name)
+          association.load_target unless association.loaded?
+          if association.target
+            target = association.target
+            target.class.update_counters(target.id, association.reflection.counter_cache_column => value)
+          end
```

Comment:
can we extract this part into `update_soft_delete_counter_cache(association)` and then call it from both methods, so the duplication is a bit lower ?

Reply:
Done.